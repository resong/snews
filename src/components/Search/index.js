import React, { Component } from "react";
import PropTypes from "prop-types";

class Search extends Component {
  componentDidMount() {
    if (this.input) {
      this.input.focus();
    }
  }

  render() {
    const { value, onChange, onSubmit, children } = this.props;

    return (
      <form onSubmit={onSubmit}>
        <input
          type="text"
          value={value}
          onChange={onChange}
          ref={node => {
            this.input = node;
          }}
        />
        <button type="submit" className="button-visible">
          {children}
        </button>
      </form>
    );
  }
}

Search.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired
};

export default Search;
