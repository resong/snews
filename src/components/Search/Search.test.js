import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Search from "../Search";

Enzyme.configure({ adapter: new Adapter() });

describe("Search", () => {
  let onChange, onSubmit, searchForm;

  beforeAll(() => {
    onChange = jest.fn();
    onSubmit = jest.fn();
    searchForm = (
      <Search onChange={onChange} onSubmit={onSubmit}>
        Search
      </Search>
    );
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(searchForm, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(searchForm);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has an input and a button", () => {
    const input = shallow(searchForm).find("input");
    expect(input).toBeDefined();

    const btn = shallow(searchForm).find("button");
    expect(btn).toBeDefined();
  });

  test("calls onChange when input changes", () => {
    const input = shallow(searchForm).find("input");
    input.simulate("change", "Different Input");
    expect(onChange).toBeCalledWith("Different Input");
  });

  test("calls onSubmit when search text is submitted", () => {
    const form = shallow(searchForm).find("form");
    form.simulate("submit");
    expect(onSubmit).toHaveBeenCalledTimes(1);
  });
});
