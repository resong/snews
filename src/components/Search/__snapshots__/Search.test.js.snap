// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has a valid snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value={undefined}
  />
  <button
    className="button-visible"
    type="submit"
  >
    Search
  </button>
</form>
`;
